src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> nodeOfReplacedValue = getNodeByIndex(index);
+        T returnValue = nodeOfReplacedValue.item;
+        nodeOfReplacedValue.item = value;
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToRemove = getNodeByIndex(index);
+        removeNodeByNode(nodeToRemove);
+        size--;
+        return nodeToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> nodeToRemove = getNodeByValue(object);
+        if (nodeToRemove == null) {
+            return false;
+        }
+        removeNodeByNode(nodeToRemove);
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void removeNodeByNode(Node<T> nodeToRemove) {
+        if (nodeToRemove.equals(first)) {
+            first = first.next;
+            if (first == null) {
+                last = null;
+            } else {
+                first.prev = null;
+            }
+        } else if (nodeToRemove.equals(last)) {
+            last = last.prev;
+            if (last == null) {
+                first = null;
+            } else {
+                last.next = null;
+            }
+        } else {
+            nodeToRemove.prev.next = nodeToRemove.next;
+            nodeToRemove.next.prev = nodeToRemove.prev;
+        }
+    }
+
+    private Node<T> getNodeByValue(T value) {
+        if (size == 0) {
+            return null;
+        }
+        Node<T> currentNode = first;
+        for (int i = 1; i < size + 1; i++) {
+            if ((currentNode.item == value)
+                    || (currentNode.item != null && currentNode.item.equals(value))) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        indexValidation(index);
+        int iterator;
+        Node<T> currentElement;
+        if (index < size / 2) {
+            iterator = 0;
+            currentElement = first;
+            while (iterator != index) {
+                currentElement = currentElement.next;
+                iterator++;
+            }
+        } else {
+            iterator = size - 1;
+            currentElement = last;
+            while (iterator != index) {
+                currentElement = currentElement.prev;
+                iterator--;
+            }
+        }
+        return currentElement;
+    }
+
+    private int indexValidation(int index) {
```

Comment:
lets make this method void and think about how we could simplify this `if (ZERO_INDEX < index && index < size || index == 0)`